client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -1,301 +1,95 @@
 
-const getSiteSearchEndpoint = ( blogId: string ) =>
-	`https://public-api.wordpress.com/rest/v1.1/sites/${ blogId }/search`;
-
-/**
- *
- * @param pluginSlug
- * @param iconsArray
- * @returns An object containing a resolution -> icon url map
- */
-const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
-	type Icon = {
-		resolution: string;
-		filename: string;
-		location: string;
-		revision: string;
-	};
-	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
-	return Object.values( icons ).reduce(
-		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
-			prev[
-				resolution
-			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
-			return prev;
-		},
-		{}
-	);
-};
-
-const createAuthorUrl = ( headerAuthor: string, headerAuthorUri: string ) =>
-	`<a href="${ headerAuthorUri }">${ headerAuthor }</a>`;
-
-const mapStarRatingToPercent = ( starRating: number ) => ( starRating / 5 ) * 100;
+// /**
+//  *
+//  * @param pluginSlug
+//  * @param iconsArray
+//  * @returns An object containing a resolution -> icon url map
+//  */
+// const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
+// 	type Icon = {
+// 		resolution: string;
+// 		filename: string;
+// 		location: string;
+// 		revision: string;
+// 	};
+// 	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
+// 	return Object.values( icons ).reduce(
+// 		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
+// 			prev[
+// 				resolution
+// 			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
+// 			return prev;
+// 		},
+// 		{}
+// 	);
+// };
+
+// const createAuthorUrl = ( headerAuthor: string, headerAuthorUri: string ) =>
+// 	`<a href="${ headerAuthorUri }">${ headerAuthor }</a>`;
+
+const mapStarRatingToPercent = ( starRating?: number ) => ( starRating ?? 0 / 5 ) * 100;
 
 const mapIndexResultsToPluginData = ( results: ESHits ): Plugin[] => {
+	if ( ! results ) return [];
 	return results.map( ( { fields: hit } ) => {
-		return {
-			name: hit.title_en, // TODO: add localization
-			slug: hit.slug,
-			version: hit.stable_tag,
-			author: createAuthorUrl(
-				hit[ 'meta.header_author.value' ],
-				hit[ 'meta.header_author_uri.value' ]
-			),
-			author_profile: hit[ 'meta.header_author_uri.value' ], // TODO: get author profile URL
-			tested: hit.tested,
-			rating: mapStarRatingToPercent( hit.rating ),
-			num_ratings: hit.num_ratings,
-			support_threads: hit.support_threads,
-			support_threads_resolved: hit.support_threads_resolved,
-			active_installs: hit.active_installs,
-			last_updated: hit[ 'meta.last_updated.value' ],
-			short_description: hit.excerpt_en, // TODO: add localization
-			icons: createIconsObject( hit.slug, hit[ 'meta.assets_icons.value' ] ),
+		const slug = hit[ 'title.default' ].toLowerCase().replace( /\s/g, '-' );
```

Comment:
This is not correct. For example Yoast SEO has `wordpress-seo` as slug. If it's something temporary let's add a comment.

Reply:
Absolutely! This is an interim/temporary solution while `slug` is provided by the response as it is not provided at the moment. I will add a clear comment in the code indicating it